### ISSUE TITLE
Enable combining targets for scoring

### ISSUE THREAD

**Issue by espertus** [espertus/jacquard#2]
If I run a single `CheckStyleGrader` over multiple targets in a single call to `Grader.grade(List<Target>)`, each target can earn (or lose) the maximum score.

For example, consider:
```java
List<Target> targets = Target.fromPathStrings("src/main/java/student/",
        "Class1.java", "Class2.java");
CheckstyleGrader checkstyleGrader = new CheckstyleGrader(
        "config/checkstyle-rules.xml",
        1.0,
        10.0);
List<Result> results = checkstyleGrader.grade(targets);
```

Because there are two targets, Checkstyle is worth 20 points, not 10. This violates the
Principle of Least Astonishment.

The same thing happens with other graders.

There are workarounds, such as constructing multiple graders or postprocessing the returned `List<Result>`, but there should be a better way.

This applies to `jacquard-examples/hw1`.

**Comment by espertus** [comment]
I fixed this in checkstyle with commit fd3f08e.

**Comment by espertus** [comment]
I fixed it in PMD with commit [64da97c](https://github.com/espertus/jacquard/commit/64da97c8617dc3440e110ffd06e50759a1fd023b).
